### PR TITLE
docs: improve branchConcurrentLimit documentation

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Explain that raising `branchConcurrentLimit` impacts Renovate's rebase performance
- Explain that `config:base` limits the amount of open branches at once
- Add advice from @rarkins https://github.com/renovatebot/renovate/discussions/11866#discussioncomment-1378209
- Use one sentence per line pattern

## Context:

I saw somebody getting a bit confused (#11866) on why Renovate takes so long to rebase their branches, so that's a sign that we should improve our documentation and explain.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -263,6 +264,15 @@ Example config:
...
 ```
 
+Warning: increasing the `branchConcurrentLimit` increases the time it takes for Renovate to complete rebasing a PR branch when it's behind the target branch.
+If you find that Renovate is too slow when rebasing out-of-date branches, decrease the `branchConcurrentLimit`.
+
+If you have too many concurrent branches which rebase themselves each run, Renovate can take a lot of time to rebase.
+Solutions:
+
+- Decrease the concurrent branch limit (note: this won't go and delete any existing, so won't have an effect until you either merge or close existing ones manually)
+- Remove automerge and/or automatic rebasing (set `rebaseWhen` to `conflicted`). However if you have branch protection saying PRs must be up to date then it's not ideal to remove automatic rebasing.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:274]: Not sure if we need this advice or not, I've basically copy/pasted the comment from the maintainer into here. But maybe that's too much information dump?

Maybe we could add this in the FAQ in a section like "Help Renovate rebases are really slow" instead?
Or maybe we don't need this explained at all....

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -248,10 +248,11 @@ Configuring this to `true` means that Renovate will detect and apply the default
 
 ## branchConcurrentLimit
 
-By default, Renovate won't enforce any concurrent branch limits. If you want the same limit for both concurrent branches
-and concurrent PRs, then just set a value for `prConcurrentLimit` and it will be reused for branch calculations too.
-However, if you want to allow more concurrent branches than concurrent PRs, you can configure both values (
-e.g. `branchConcurrentLimit=5` and `prConcurrentLimit=3`).
+By default, Renovate won't enforce any concurrent branch limits.
+The `config:base` preset that many extend from limits the amount of concurrent branches.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:252]:
```suggestion
The `config:base` preset that many extend from limits the amount of concurrent branches to 20, but in most cases we would recommend lower values such as 5 or 10.
```

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -263,6 +264,15 @@ Example config:
 }
 ```
 
+Warning: increasing the `branchConcurrentLimit` increases the time it takes for Renovate to complete rebasing a PR branch when it's behind the target branch.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:267]:
```suggestion
Warning: Leaving PRs/branches as unlimited or as a high number increases the time it takes for Renovate to process a repository.
```

File: docs/usage/configuration-options.md (hunk 1 above)
Comment (rarkins) [reply docs/usage/configuration-options.md:274]: it's ok to leave

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md (hunk 1 above)
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:274]:
```suggestion
- Remove automerge and/or automatic rebasing (set `rebaseWhen` to `conflicted`). However if you have branch protection saying PRs must be up to date then it's not ideal to remove automatic rebasing
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**